e2e/tests/attendance-tests.spec.ts

```diff
@@ -1,360 +1,278 @@
-    await page.clock.install({ time: FIXED_DATE });
-  });
+  const dateField = page.getByLabel("Date");
+  await expect(dateField).toHaveValue("1/23/2025");
 
-  test("Record attendance for one activity", async ({ page }) => {
-    await page.getByRole("navigation").getByText("Attendance").click();
-    await page.getByRole("button", { name: "Record" }).click();
+  await dateField.fill("12/15/2024");
 
-    /*
-      Verify the date field displays the current date
-    */
-    const fixedDate = new Date(FIXED_DATE);
-    const formattedDateRegex = new RegExp(
-      `${fixedDate.getMonth() + 1}/${fixedDate.getDate()}/${fixedDate.getFullYear()}|` +
-        `${(fixedDate.getMonth() + 1).toString().padStart(2, "0")}/${fixedDate.getDate().toString().padStart(2, "0")}/${fixedDate.getFullYear()}`,
-    );
-    const dateField = page.getByLabel("Date");
-    await expect(dateField).toHaveValue(formattedDateRegex);
+  // FIXME: A simple .click() does not trigger the action and we don’t know why.
+  await page.getByText("School Class 2B").dispatchEvent("click");
 
-    /*
-      Allow backdated editing for the date field
-    */
-    const backdatedDate = "12/15/2024";
-    await dateField.fill(backdatedDate);
-    await expect(dateField).toHaveValue(backdatedDate);
+  await expect(
+    page.getByRole("heading", { name: "School Class 2B" }),
+  ).toBeVisible();
 
-    /*
-      Verify list of classes is displayed
-    */
-    await page.waitForSelector("mat-card-header");
-    const classList = page.locator("mat-card-header");
-    const count = await classList.count();
-    expect(count).toBeGreaterThan(0);
+  await expect(page.getByText("1 / 6")).toBeVisible();
+  await page.getByRole("button", { name: "Present" }).click();
 
-    /*
-      Mark attendance for participants dynamically
-    */
-    const firstClass = page.locator("mat-card-header").first();
-    await firstClass.evaluate((el: HTMLElement) => el?.click()); // Force the DOM click
-
-    let totalParticipants = 0;
-
-    const pageIndicator = page.locator("text=/\\d+\\s\\/\\s\\d+/"); // "1 / 5" pattern
-    // Extract total participants from "x / y" format
-    if (await pageIndicator.isVisible()) {
-      const text = await pageIndicator.textContent();
-      const match = text?.match(/(\d+)\s\/\s(\d+)/); // Extract current/total numbers
-      if (match) {
-        totalParticipants = parseInt(match[2]); // Total participants
-      }
-    }
-    const ReviewDetailsBtn = page.getByRole("button", {
-      name: "Review Details",
-    });
-
-    // Loop to mark attendance dynamically until the "Review Details" button is visible
-    while (!(await ReviewDetailsBtn.isVisible())) {
-      const randomIndex = Math.floor(Math.random() * 4); // Random index for attendance buttons
-      const attendanceButton = page.getByRole("paragraph").nth(randomIndex);
-      await attendanceButton.click();
-      await page.waitForTimeout(1000); // Wait for the next participant to load
```

Comment:
We should try to avoid this fixed time-outs at all costs because they are guaranteed to slow down the tests. (With 6 attendees we’re just waiting here for 6s). It’s usually possible to rewrite this as an expectation that is only waited for until it succeeds.

To work around the loading issue I used
```typescript
await expect(page.getByRole("button", { name: "Absent" })).toHaveCount(1
````